string[] commands
int16[] speeds
